### PR TITLE
feat: support ENV variable in configuration

### LINKED ISSUE #2675: feat: support ENV variable in `conf/config.yaml`
Here is an example, we can take a look:

```yaml
apisix:
  node_listen: ${APISIX_NODE_LISTEN} # APISIX listening port
```

### PR DESCRIPTION
**Pull request by spacewander**
Close #2675.

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible?

### CODE REVIEW

**Comment by nic-chen** [comment]
nice feature.

**Review by nic-chen — APPROVED** [review]
LGTM

File: .travis/apisix_cli_test.sh [hunk 1]
```diff
@@ -144,6 +144,50 @@ fi
 
 echo "passed: change default env"
 
+# support environment varialbes
```
Comment (agile6v) [inline .travis/apisix_cli_test.sh:147]: `varialbes` => `variables`

Comment (agile6v) [reply .travis/apisix_cli_test.sh:147]: There are some typos below.

**Review by agile6v — APPROVED** [review]

File: .travis/apisix_cli_test.sh (hunk 1 above)
Comment (spacewander) [reply .travis/apisix_cli_test.sh]: @agile6v 
Typo fixed.

**Pull request merged**